pkgs/applications/virtualization/podman/default.nix

```diff
@@ -57,21 +58,27 @@ let
       catatonit # added here for the pause image and also set in `containersConf` for `init_path`
       netavark
       slirp4netns
+      passt
     ];
   };
 in
 buildGoModule rec {
   pname = "podman";
-  version = "4.9.3";
+  version = "5.0.1";
 
   src = fetchFromGitHub {
     owner = "containers";
     repo = "podman";
     rev = "v${version}";
-    hash = "sha256-PdAXcXtc/Jl3ttWWB6TciiOwWescJ51Glhf2ZhOw550=";
+    hash = "sha256-XgLrPLswLmaB9FYXKEMLP+7KT/OY50z3JKz8DvMLrEE=";
   };
 
   patches = [
+    (substituteAll {
+      src = ./hardcode-paths.patch;
+      inherit crun runc gvisor youki conmon;
```

Comment:
I think we should try to limit this to what is useful. I thin kcrun is in almost every case better than runc, so we can drop that.

see https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/building_running_and_managing_containers/selecting-a-container-runtime_building-running-and-managing-containers#con_the-crun-container-runtime_selecting-a-container-runtime

Reply:
Why would we want to limit this - to reduce the number of dependencies?

But, if `runc` is obsolete, why would upstream still continue supporting it?

Since upstream continues supporting it, wouldn't that imply that it is valuable to have it as an option?

Also, if we don't want to support `runc`, shouldn't we delete it from nixpkgs?

Reply:
> Why would we want to limit this - to reduce the number of dependencies?

Yeah, we increased the dependencies with this update a lot.

> But, if `runc` is obsolete, why would upstream still continue supporting it?

Maybe there are certain usecases when it is useful which are not relevant to nixpkgs?

Reply:
Would it make sense to make these dependencies (`runc`, `gvisor`, `youki`) optional?

Reply:
Yeah, I think so. I think we could right now overwrite them with null, but that is a bit dirty.

Reply:
https://github.com/NixOS/nixpkgs/pull/310182 does this. I left `runc` in by default on Linux, because nixpkgs actually has a test that podman works with that runtime. But, it can now be disabled.